contracts/transfer/circuits/src/dusk_contract/withdraw_from_obfuscated.rs

```diff
@@ -0,0 +1,362 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use dusk_plonk::constraint_system::ecc::scalar_mul::fixed_base::scalar_mul;
+use dusk_plonk::prelude::*;
+use crate::gadgets::range::range;
+use dusk_plonk::jubjub::{
+    AffinePoint, GENERATOR_EXTENDED, GENERATOR_NUMS_EXTENDED
+};
+use plonk_gadgets::AllocatedScalar;
+use anyhow::{Error, Result};
+
+/// The circuit responsible for creating a zero-knowledge proof
+/// for a 'withdraw from obfuscated contract' transaction.
+#[derive(Debug, Default, Clone)]
+pub struct WithdrawFromContractObfuscatedCircuit {
+    /// Spend Value within Pedersen commitment
+    pub spend_commitment_value: Option<BlsScalar>,
+    /// Spend Blinder within Pedersen commitment
+    pub spend_commitment_blinder: Option<BlsScalar>,
+    /// Spend Pedersen Commitment 
+    pub spend_commitment: Option<AffinePoint>,
+    /// Message Value within Pedersen commitment
+    pub message_commitment_value: Option<BlsScalar>,
+    /// Message Blinder within Pedersen commitment
+    pub message_commitment_blinder: Option<BlsScalar>,
+    /// Message Pedersen Commitment 
+    pub message_commitment: Option<AffinePoint>,
+    /// Note Value within Pedersen commitment
+    pub note_commitment_value: Option<BlsScalar>,
+    /// Note Blinder within Pedersen commitment
+    pub note_commitment_blinder: Option<BlsScalar>,
+    /// Note Pedersen Commitment 
+    pub note_commitment: Option<AffinePoint>,
```

Comment:
Is there any structure that holds more than one of these fields inside? If that's the case, you should add the structure here instead of its fields.